primerpedia.js

```diff
@@ -196,9 +200,33 @@ function getQueryVariable(parameter) {
 	return null;
 }
 
+function setSearchButtonEnabledState() {
+	var searchTermInputElementValid = searchTermInputElement !== undefined &&
+		searchTermInputElement !== null &&
+		searchTermInputElement instanceof HTMLInputElement;
+
+	var searchButtonAllowedTypes = ["submit", "button"];
+
+	var searchButtonValid = searchButton !== undefined &&
+		searchButton !== null &&
+		searchButton instanceof HTMLInputElement &&
+		searchButtonAllowedTypes.indexOf(searchButton.getAttribute("type")) > -1;
+
+	if(searchTermInputElementValid && searchButtonValid) {
```

Comment:
While I comment the care to avoid errors, I think the toll to pay in this case is too great: the lines setting up protection against possible errors are more numerous than the actual code of the function!

Why wouldn't the simpler `if(searchTermInputElement && searchButton)` suffice?

Reply:
On a second glance I also think it would suffice to just check if it's an instance of the appropriate type (HTMLElement / HTMLInpurtElement). As I said - in most cases this code will not be hit before `onload` ran.

I'm not ok with using code like `if(searchTermInputElement && searchButton)` ever (if these variables are not booleans) because it just ensures that these both variables are not falsy. However we require them to be of a specific value (or more precise - a specific instance).

Reply:
> I also think it would suffice to just check if it's an instance of the appropriate type (HTMLElement / HTMLInpurtElement).

Reducing the condition to a single test per element seems like a sensible middle-ground. It may not cover every single edge case, but I'd really prefer not to make the code too verbose without a strong justification, as every line added reduces its approachability and readability.

Please go ahead :)